scripts/irods/test/test_irepl.py

```diff
@@ -212,6 +212,104 @@ def test_irepl_a_updates_stale_zero_byte_replicas__issue_6285(self):
             lib.remove_resource(self.admin, roots[0])
             lib.remove_resource(self.admin, roots[1])
 
+    def test_server_honors_server_config_option_checksum_read_buffer_size_in_bytes__issue_7947(self):
+
+        def get_checksum(data_object, replica_number, expected_length_of_checksum=49):
+            gql = "select DATA_CHECKSUM where COLL_NAME = '{0}' and DATA_NAME = '{1}' and DATA_REPL_NUM = '{2}'"
+            checksum, err, ec = self.admin.run_icommand(['iquest', '%s', gql.format(self.admin.session_collection, data_object, replica_number)])
+            self.assertEqual(ec, 0)
+            self.assertEqual(len(err), 0)
+            self.assertEqual(len(checksum.strip()), expected_length_of_checksum)
+            return checksum.strip()
+
+        object_name = 'test_iput_with_updated_checksum_read_buffer_size_in_bytes__issue_7947'
+        server_config_filename = paths.server_config_path()
+
+        # load server_config.json to inject a new rule base
+        with open(server_config_filename) as f:
+            svr_cfg = json.load(f)
+
+        # update the checksum_read_buffer_size_in_bytes to 10 MiB
+        svr_cfg['advanced_settings']['checksum_read_buffer_size_in_bytes'] = 10*1024*1024
+
+        # dump to a string to repave the existing server_config.json
+        new_server_config = json.dumps(svr_cfg, sort_keys=True, indent=4, separators=(',', ': '))
+
+        ufs_resources = ['issue_7947_ufs1', 'issue_7947_ufs2']
+
+        try:
+            # create two resources
+            for resc_name in ufs_resources:
+                lib.create_ufs_resource(self.admin, resc_name)
+
+            # create a 100 MiB file
+            file_size = 100*1024*1024
```

Comment:
Thinking on this some more, we should probably include tests for the following:

- Smallest valid size (1 byte)
- Largest valid size (2^31-1(?))
- buffer size == size of file
- buffer size == size of file - 1
- buffer size == size of file + 1

We can punt on this by creating a new issue for it. Up to you.

Reply:
Aren't we asking for problems if we try a buffer size of 2 GiB?

Reply:
Mmmm probably. We don't have to do that one. Or maybe we can skip the test based on the amount of memory available on the machine (might be running in a k8s pod!).

Reply:
I added it in with a check on each loop to make sure we have at least twice the memory and put the 1 byte buffer by itself with a much smaller file.